Checkout

norberttech opened the pull request.

norberttech committed:
Added New Checkout command

norberttech committed:
Added Change Shipping Address command

norberttech committed:
Removed useless code

norberttech committed:
Added basic country ISO2 code validation

potfur commented on src/Dumplie/Domain/Customer/Checkout.php:
```diff
@@ -0,0 +1,66 @@
+     * @param CartId $cartId
+     * @param Address $billingAddress
+     */
+    public function __construct(CartId $cartId, Address $billingAddress)
```
It's good that _billing address_ is required, but it would be nice to be able to change after creation.

treku commented on src/Dumplie/Application/Command/Customer/ChangeShippingAddress.php:
```diff
@@ -0,0 +1,104 @@
+     * @param string $city
+     * @param string $countryIso2Code
+     */
+    public function __construct(string $cartId, string $name, string $street, string $postCode, string $city, string $countryIso2Code)
```
Why do we use so many variables here? Isn't it better to just use variabile of type Address and get rid of all private variables in this class?

potfur replied:
This is a command, and as such it is basically a DTO, no need for VOs there.

treku commented on src/Dumplie/Application/Command/Customer/NewCheckout.php:
```diff
@@ -0,0 +1,104 @@
+     * @param string $city
+     * @param string $countryIso2Code
+     */
+    public function __construct(string $cartId, string $name, string $street, string $postCode, string $city, string $countryIso2Code)
```
(copy of the comment for ChangeShippingAddress class)
Why do we use so many variables here? Isn't it better to just use variabile of type Address and get rid of all private variables in this class?

norberttech committed:
Added possibility to change billing address, fixed few small issues

norberttech commented:
So I just added possibility to change billing address for checkout. I also moved common address parts for commands to trait to avoid code duplication. I also changed order for code execution with transactions.
